### ISSUE TITLE
Session problem in test_profile.

### ISSUE THREAD

**Issue by Arkkienkeli** [cytomining/DeepProfiler#211]
test_profiling.py::test_configure ERROR
test_profiling.py::test_profile ERROR

@jccaicedo We need to check this out.

```    @pytest.fixture(scope="function")
    def checkpoint(config, dataset):
        crop_generator = importlib.import_module(
            "plugins.crop_generators.{}".format(config["train"]["model"]["crop_generator"])) \
            .GeneratorClass
        profile_crop_generator = importlib.import_module(
            "plugins.crop_generators.{}".format(config["train"]["model"]["crop_generator"])) \
            .SingleImageGeneratorClass
        dpmodel = importlib.import_module("plugins.models.{}".format(config["train"]["model"]["name"])) \
            .ModelClass(config, dataset, crop_generator, profile_crop_generator)
        dpmodel.feature_model.compile(dpmodel.optimizer, dpmodel.loss)
        filename = os.path.join(config["paths"]["checkpoints"], config["profile"]["checkpoint"])
>       dpmodel.feature_model.save_weights(filename)

tests/deepprofiler/learning/test_profiling.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
deep36/lib/python3.6/site-packages/keras/engine/network.py:1121: in save_weights
    saving.save_weights_to_hdf5_group(f, self.layers)
deep36/lib/python3.6/site-packages/keras/engine/saving.py:572: in save_weights_to_hdf5_group
    weight_values = K.batch_get_value(symbolic_weights)
deep36/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:2420: in batch_get_value
    return get_session().run(ops)
deep36/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:199: in get_session
    [tf.is_variable_initialized(v) for v in candidate_vars])
deep36/lib/python3.6/site-packages/tensorflow/python/client/session.py:929: in run
    run_metadata_ptr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tensorflow.python.client.session.Session object at 0x7fdf0b72ed30>, handle = None
fetches = [<tf.Tensor 'IsVariableInitialized_99:0' shape=() dtype=bool>, <tf.Tensor 'IsVariableInitialized_100:0' shape=() dtype...'IsVariableInitialized_103:0' shape=() dtype=bool>, <tf.Tensor 'IsVariableInitialized_104:0' shape=() dtype=bool>, ...]
feed_dict = None, options = None, run_metadata = None

    def _run(self, handle, fetches, feed_dict, options, run_metadata):
      """Perform either run or partial_run, depending the presence of `handle`."""
    
      def _feed_fn(feed, feed_val):
        for tensor_type, _, feed_fn, _ in _REGISTERED_EXPANSIONS:
          if isinstance(feed, tensor_type):
            return feed_fn(feed, feed_val)
        raise TypeError('Feed argument %r has invalid type %r' % (feed,
                                                                  type(feed)))
    
      # Check session.
      if self._closed:
>       raise RuntimeError('Attempted to use a closed Session.')
E       RuntimeError: Attempted to use a closed Session.
```

**Comment by jccaicedo** [comment]
Thanks for reporting this!

If I run pytest for all unit tests, the errors show up. If I run pytest only for the profiling functionalities, there are no errors.

Apparently the errors come from the fact that the tests run training and profiling at the same time, and there is a conflict with the sessions. At the moment, it's an issue of how tests are running rather than a problem with the actual functionalities. I'll have a look at this!

**Comment by Arkkienkeli** [comment]
Fixed 18aff73cf56b5faa295b94dcf128f214530ffdc4
